kubernetes/RemoteResource/resource.yaml

```diff
@@ -233,13 +233,26 @@ items:
                             oneOf:
                               - required: [url]
                               - required: [uri]
+                              - required: [git]
                             properties:
                               url:
                                 type: string
                                 format: uri
                               uri:
                                 type: string
                                 format: uri
+                              git: 
+                                type: object
+                                required: [gittype, repo, branch, pattern]
```

Comment:
some suggestions on what might be better naming for the user (open for discussion and doesnt mean the backing code needs to change, just want it clear to the user what they are providing us)
`gittype -> provider`: Since we are already under the git section, shouldnt need to use git in the property name again. provider should tell the user to give us the name of who they use to store their source code.
`pattern -> filePath`: while this makes it less clear that we are allowing patterns, i think it makes more sense since it tells the user what this field is being used for. We can/should include in the docs that they are allowed to use wildcarding patterns in the filePath.